twenty-first/src/shared_math/polynomial.rs

```diff
@@ -1074,10 +1067,67 @@ impl<PFElem: FiniteField> Sub for Polynomial<PFElem> {
     }
 }
 
+impl<BFieldElement: FiniteField> Polynomial<BFieldElement> {
+    // Division in 𝔽_p[X], not 𝔽_{p^e} ≅ 𝔽[X]/p(x).
+    pub fn xgcd(
+        mut x: Polynomial<BFieldElement>,
+        mut y: Polynomial<BFieldElement>,
+    ) -> (
+        Polynomial<BFieldElement>,
+        Polynomial<BFieldElement>,
+        Polynomial<BFieldElement>,
+    ) {
+        let mut a_factor = Polynomial::new(vec![BFieldElement::one()]);
+        let mut a1 = Polynomial::new(vec![BFieldElement::zero()]);
+        let mut b_factor = Polynomial::new(vec![BFieldElement::zero()]);
+        let mut b1 = Polynomial::new(vec![BFieldElement::one()]);
```

Comment:
```suggestion
impl<FFelem: FiniteField> Polynomial<FFelem> {
    pub fn xgcd(
        mut x: Polynomial<FFelem>,
        mut y: Polynomial<FFelem>,
    ) -> (
        Polynomial<FFelem>,
        Polynomial<FFelem>,
        Polynomial<FFelem>,
    ) {
        let mut a_factor = Polynomial::new(vec![FFelem::one()]);
        let mut a1 = Polynomial::new(vec![FFelem::zero()]);
        let mut b_factor = Polynomial::new(vec![FFelem::zero()]);
        let mut b1 = Polynomial::new(vec![FFelem::one()]);
```
No reason to limit this implementation to polynomials over $\mathbb{F}_p$ for one specific $p$.

Reply:
Done. I'm OK with this suggestion now that I have added tests for both B field and X field polynomials.